Sources/TAPS/TLSClientService.swift

```diff
@@ -0,0 +1,175 @@
+    public var clientCertificate: ClientCertificate?
+    
+    // MARK: - Advanced TLS Options
+    
+    /// Cipher suites (nil means system default)
+    public var cipherSuites: [String]?
+    
+    /// Application Layer Protocol Negotiation (ALPN) protocols
+    public var alpnProtocols: [String]
+    
+    /// Whether to enable session resumption
+    public var enableSessionResumption: Bool
+    
+    /// Custom certificate validation callback
+    public var customCertificateValidation: (@Sendable (TLSCertificateInfo) async -> Bool)?
+    
+    public init(
+        connectionTimeout: Duration = .seconds(30),
+        keepAlive: Bool = false,
+        noDelay: Bool = true,
+        minimumTLSVersion: TLSVersion = .v1_2,
+        maximumTLSVersion: TLSVersion? = nil,
+        serverHostname: String? = nil,
+        certificateVerification: CertificateVerification = .fullVerification,
+        trustedCertificates: [String] = [],
+        clientCertificate: ClientCertificate? = nil,
+        cipherSuites: [String]? = nil,
+        alpnProtocols: [String] = [],
+        enableSessionResumption: Bool = true,
+        customCertificateValidation: (@Sendable (TLSCertificateInfo) async -> Bool)? = nil
+    ) {
+        self.connectionTimeout = connectionTimeout
+        self.keepAlive = keepAlive
+        self.noDelay = noDelay
+        self.minimumTLSVersion = minimumTLSVersion
+        self.maximumTLSVersion = maximumTLSVersion
+        self.serverHostname = serverHostname
+        self.certificateVerification = certificateVerification
+        self.trustedCertificates = trustedCertificates
+        self.clientCertificate = clientCertificate
+        self.cipherSuites = cipherSuites
+        self.alpnProtocols = alpnProtocols
+        self.enableSessionResumption = enableSessionResumption
+        self.customCertificateValidation = customCertificateValidation
+    }
+    
+    public static var defaultParameters: TLSClientParameters {
+        return TLSClientParameters()
+    }
+}
+
+// MARK: - TLS Configuration Types
+
+/// TLS protocol versions
+public enum TLSVersion: String, Sendable, CaseIterable {
+    case v1_0 = "TLSv1.0"
+    case v1_1 = "TLSv1.1"
+    case v1_2 = "TLSv1.2"
+    case v1_3 = "TLSv1.3"
+}
+
+/// Certificate verification modes
+public enum CertificateVerification: Sendable {
+    /// Full certificate chain and hostname verification (recommended)
+    case fullVerification
+    /// Verify certificate chain but skip hostname verification
+    case certificateOnly
+    /// Skip all certificate verification (dangerous - for testing only)
+    case noVerification
+}
+
+/// Client certificate configuration for mTLS
+public struct ClientCertificate: Sendable {
+    /// Certificate chain in PEM format
+    public let certificateChain: String
+    /// Private key in PEM format
+    public let privateKey: String
+    /// Private key passphrase (if encrypted)
+    public let privateKeyPassphrase: String?
```

Comment:
I hope we can skip this, and delegate the decryption of private keys to the call site